### PR TITLE
[Feat] add Lock Package

### PR DESCRIPTION
**Pull request by dongjiang1989**
### What type of PR is this?
/kind feature

- [x] Unittest
- [x] Coverage 80%+
- [x] Build Success

### What this PR does / why we need it:
 Add  Lock Package: support file lock and memory lock

### CODE REVIEW

File: pkg/lock/memory.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"runtime"
+	"sync/atomic"
+)
+
+// MemoryLock implements sync/Locker, default 0 indicates an unlocked memory.
+type MemoryLock struct {
+	_flag uint32
+}
+
```
Comment (imlense) [inline pkg/lock/memory.go:28]: add `NewMemoryLock` func

Comment (imlense) [reply pkg/lock/memory.go:28]: THKS! Done. Add `NewMemoryLock` function

Comment (dongjiang1989) [reply pkg/lock/memory.go:28]: Thinks for you review! Done. Add `NewMemoryLock` create function

**Review by imlense — COMMENTED** [review]
LGTM

**Pull request merged**